logstreams/src/main/java/io/zeebe/logstreams/impl/log/BufferedLogStreamReader.java

```diff
@@ -45,13 +44,15 @@
   private ByteBuffer byteBuffer;
   private int bufferOffset;
 
-  public BufferedLogStreamReader(final LogStorage logStorage) {
-    this();
-    wrap(logStorage);
-  }
-
-  public BufferedLogStreamReader() {
+  BufferedLogStreamReader(final LogStorage logStorage) {
```

Comment:
Could be simplified with getting the reader directly instead of the storage - but I plan on refactoring this so feel free to ignore this suggestion